diff --git a/node_modules/podverse-shared/dist/htmlHelpers.js b/node_modules/podverse-shared/dist/htmlHelpers.js
index ff3ffa5..83d4f1e 100644
--- a/node_modules/podverse-shared/dist/htmlHelpers.js
+++ b/node_modules/podverse-shared/dist/htmlHelpers.js
@@ -23,7 +23,7 @@ exports.decodeHTMLString = function (text) {
 exports.filterHTMLElementsFromString = function (html) {
     if (html) {
         // eslint-disable-next-line max-len
-        var finalHtml = html.replace(/<audio.*>.*?<\/audio>|<video.*>.*?<\/video>|<iframe.*>.*?<\/iframe>|<img.*>.*?<\/img>|<img.*>/gi, '');
+        var finalHtml = html.replace(/<audio.*>.*?<\/audio>|<video.*>.*?<\/video>|<iframe.*>.*?<\/iframe>|<img.*>.*?<\/img>|<img.*?\/>|<img>/gi, '');
         return finalHtml;
     }
     return html;
