diff --git a/node_modules/react-native-track-player/ios/RNTrackPlayer/Models/Track.swift b/node_modules/react-native-track-player/ios/RNTrackPlayer/Models/Track.swift
index ea5a302..c061d32 100644
--- a/node_modules/react-native-track-player/ios/RNTrackPlayer/Models/Track.swift
+++ b/node_modules/react-native-track-player/ios/RNTrackPlayer/Models/Track.swift
@@ -11,7 +11,8 @@ import MediaPlayer
 import AVFoundation
 import SwiftAudioEx
 
-class Track: AudioItem, TimePitching, AssetOptionsProviding {
+/* Podverse patch https://gist.github.com/gilbertl/82e3043fa5632ef653b274a2b86f59f2 */
+class Track: AudioItem, TimePitching, AssetOptionsProviding, InitialTiming {
     let url: MediaURL
 
     @objc var title: String?
@@ -27,6 +28,9 @@ class Track: AudioItem, TimePitching, AssetOptionsProviding {
     let pitchAlgorithm: String?
     var isLiveStream: Bool?
 
+    /* Podverse patch https://gist.github.com/gilbertl/82e3043fa5632ef653b274a2b86f59f2 */
+    var initialTime: TimeInterval = 0.0
+
     var album: String?
     var artwork: MPMediaItemArtwork?
 
@@ -60,9 +64,21 @@ class Track: AudioItem, TimePitching, AssetOptionsProviding {
         self.artworkURL = MediaURL(object: dictionary["artwork"])
         self.isLiveStream = dictionary["isLiveStream"] as? Bool
 
+        /* Podverse patch https://gist.github.com/gilbertl/82e3043fa5632ef653b274a2b86f59f2 */
+        let initialTime = dictionary["iosInitialTime"] as? Double
+        if let x = initialTime {
+            self.initialTime = x
+        }
+
         self.originalObject = self.originalObject.merging(dictionary) { (_, new) in new }
     }
 
+    /* Podverse patch https://gist.github.com/gilbertl/82e3043fa5632ef653b274a2b86f59f2 */
+    // MARK: - InitialTiming Protocol
+    func getInitialTime() -> TimeInterval {
+        return initialTime
+    }
+
     // MARK: - AudioItem Protocol
 
     func getSourceUrl() -> String {
