diff --git a/node_modules/react-native-video-controls/VideoPlayer.js b/node_modules/react-native-video-controls/VideoPlayer.js
index 7d36b4b..8fb7086 100644
--- a/node_modules/react-native-video-controls/VideoPlayer.js
+++ b/node_modules/react-native-video-controls/VideoPlayer.js
@@ -106,7 +106,7 @@ export default class VideoPlayer extends Component {
       toggleFullscreen: this._toggleFullscreen.bind(this),
       togglePlayPause: this._togglePlayPause.bind(this),
       toggleControls: this._toggleControls.bind(this),
-      toggleTimer: this._toggleTimer.bind(this),
+      toggleTimer: this._toggleTimer.bind(this)
     };
 
     /**
@@ -613,7 +613,7 @@ export default class VideoPlayer extends Component {
    * @return {float} position of seeker handle in px based on currentTime
    */
   calculateSeekerPosition() {
-    const percent = this.state.currentTime / this.state.duration;
+    const percent = this.state.duration === 0 ? 0 : this.state.currentTime / this.state.duration;
     return this.player.seekerWidth * percent;
   }
 
@@ -1022,6 +1022,15 @@ export default class VideoPlayer extends Component {
     );
   }
 
+  renderSettings() {
+    let source = require('./assets/img/settings.png');
+    return this.renderControl(
+      <Image source={source} />,
+      this.props.showSettings,
+      styles.controls.settings,
+    );
+  }
+
   /**
    * Render bottom control group and wrap it in a holder
    */
@@ -1035,6 +1044,9 @@ export default class VideoPlayer extends Component {
     const playPauseControl = this.props.disablePlayPause
       ? this.renderNullControl()
       : this.renderPlayPause();
+    const settingsControl = this.props.disableSettings
+      ? this.renderNullControl()
+      : this.renderSettings();
 
     return (
       <Animated.View
@@ -1055,6 +1067,7 @@ export default class VideoPlayer extends Component {
             {playPauseControl}
             {this.renderTitle()}
             {timerControl}
+            {settingsControl}
           </SafeAreaView>
         </ImageBackground>
       </Animated.View>
@@ -1350,6 +1363,10 @@ const styles = {
       width: 80,
       zIndex: 0,
     },
+    settings: {
+      position: 'relative',
+      zIndex: 0
+    },
     title: {
       alignItems: 'center',
       flex: 0.6,
diff --git a/node_modules/react-native-video-controls/assets/.DS_Store b/node_modules/react-native-video-controls/assets/.DS_Store
new file mode 100644
index 0000000..f2709ea
Binary files /dev/null and b/node_modules/react-native-video-controls/assets/.DS_Store differ
diff --git a/node_modules/react-native-video-controls/assets/img/.DS_Store b/node_modules/react-native-video-controls/assets/img/.DS_Store
new file mode 100644
index 0000000..fa9bc5a
Binary files /dev/null and b/node_modules/react-native-video-controls/assets/img/.DS_Store differ
diff --git a/node_modules/react-native-video-controls/assets/img/settings.png b/node_modules/react-native-video-controls/assets/img/settings.png
new file mode 100644
index 0000000..0867d85
Binary files /dev/null and b/node_modules/react-native-video-controls/assets/img/settings.png differ
diff --git a/node_modules/react-native-video-controls/assets/img/settings@2x.png b/node_modules/react-native-video-controls/assets/img/settings@2x.png
new file mode 100644
index 0000000..99866b1
Binary files /dev/null and b/node_modules/react-native-video-controls/assets/img/settings@2x.png differ
diff --git a/node_modules/react-native-video-controls/assets/img/settings@3x.png b/node_modules/react-native-video-controls/assets/img/settings@3x.png
new file mode 100644
index 0000000..66e49db
Binary files /dev/null and b/node_modules/react-native-video-controls/assets/img/settings@3x.png differ
