diff --git a/node_modules/react-native-video/android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java b/node_modules/react-native-video/android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java
index 8850610..01818ef 100644
--- a/node_modules/react-native-video/android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java
+++ b/node_modules/react-native-video/android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java
@@ -1010,6 +1010,7 @@ class ReactExoplayerView extends FrameLayout implements
                 text += "ready";
                 eventEmitter.ready();
                 onBuffering(false);
+                clearProgressMessageHandler(); // ensure there is no other message
                 startProgressHandler();
                 videoLoaded();
                 if (selectTrackWhenReady && isUsingContentResolution) {
